### PR TITLE
Add ratelimiter flags for karmada-agent

### PR DESCRIPTION
**Pull request by dddddai**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Per https://github.com/karmada-io/karmada/pull/1489#issuecomment-1072928655

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
`karmada-agent`: Introduced several flags to specify controller's capacities:
- --rate-limiter-base-delay
- --rate-limiter-max-delay
- --rate-limiter-qps
- --rate-limiter-bucket-size
```

### CODE REVIEW

File: cmd/agent/app/agent.go [hunk 1]
```diff
@@ -309,9 +313,11 @@ func registerWithControlPlaneAPIServer(controlPlaneRestConfig, clusterRestConfig
 	}
 
 	impersonatorSecret.OwnerReferences = []metav1.OwnerReference{
-		*metav1.NewControllerRef(clusterObj, clusterv1alpha1.SchemeGroupVersion.WithKind("Cluster"))}
+		*metav1.NewControllerRef(clusterObj, clusterv1alpha1.SchemeGroupVersion.WithKind("Cluster")),
+	}
 	impersonatorSecret.Data = map[string][]byte{
-		clusterv1alpha1.SecretTokenKey: clusterImpersonatorSecret.Data[clusterv1alpha1.SecretTokenKey]}
+		clusterv1alpha1.SecretTokenKey: clusterImpersonatorSecret.Data[clusterv1alpha1.SecretTokenKey],
+	}
```
Comment (RainbowMango) [inline cmd/agent/app/agent.go:320]: What's changed here?

Comment (dddddai) [reply cmd/agent/app/agent.go:320]: It's modified by my vscode extension automatically, I guess it's because I set
`gopls.formatting.gofumpt: true`

Comment (RainbowMango) [reply cmd/agent/app/agent.go:320]: Can you revoke the change? Because it will pollute the code history and bring distraction from reviewing.

Comment (dddddai) [reply cmd/agent/app/agent.go]: OK, reverted

File: cmd/controller-manager/app/controllermanager.go [hunk 2]
```diff
@@ -275,7 +275,7 @@ func startExecutionController(ctx controllerscontext.Context) (enabled bool, err
 		PredicateFunc:        helper.NewExecutionPredicate(ctx.Mgr),
 		InformerManager:      informermanager.GetInstance(),
 		ClusterClientSetFunc: util.NewClusterDynamicClientSet,
-		RatelimiterOption:    ctx.Opts.RateLimiterOptions,
+		RatelimiterOptions:   ctx.Opts.RateLimiterOptions,
```
Comment (RainbowMango) [inline cmd/controller-manager/app/controllermanager.go:278]: nice finding.

**Review by RainbowMango — APPROVED** [review]
/approve

Leave LGTM to @Garrybest or @pigletfly

**Pull request merged**